OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -18,31 +19,78 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class AttackMoveActivity : Activity
 	{
-		const int ScanInterval = 7;
-
+		readonly Func<Activity> getInner;
+		public readonly bool IsAssaultMove;
 		Activity inner;
-		int scanTicks;
+		Activity attack;
 		AutoTarget autoTarget;
+		bool moving;
 
-		public AttackMoveActivity(Actor self, Activity inner)
+		public AttackMoveActivity(Actor self, Func<Activity> getInner, bool assaultMoving = false)
 		{
-			this.inner = inner;
+			this.getInner = getInner;
 			autoTarget = self.TraitOrDefault<AutoTarget>();
+			moving = false;
+			IsAssaultMove = assaultMoving;
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (autoTarget != null && --scanTicks <= 0)
+			if (IsCanceling)
 			{
-				autoTarget.ScanAndAttack(self, true);
-				scanTicks = ScanInterval;
+				if (attack != null)
+				{
+					attack = ActivityUtils.RunActivity(self, attack);
+					return this;
+				}
+
+				if (inner != null)
+				{
+					inner = ActivityUtils.RunActivity(self, inner);
+					return this;
+				}
+
+				return NextActivity;
+			}
+
+			if (autoTarget == null)
+				return getInner();
```

Comment:
Returning an activity without first inserting it into the graph will break the graph. We also can't just queue the inner activity, since something else might already be queued right after.  I'm afraid in this case `AttackMove` will have to be just a wrapper around the inner activity, while skipping all the attack stuff.

In other words: I think it would be enough to change the condition in the line below to `if (attack == null && autoTarget != null)`

Reply:
Fixed.